tokio/src/runtime/task/inject.rs

```diff
@@ -75,15 +75,13 @@ impl<T: 'static> Inject<T> {
 
     /// Pushes a value into the queue.
     ///
-    /// Returns `Err(task)` if pushing fails due to the queue being shutdown.
-    /// The caller is expected to call `shutdown()` on the task **if and only
-    /// if** it is a newly spawned task.
-    pub(crate) fn push(&self, task: task::Notified<T>) -> Result<(), task::Notified<T>> {
+    /// This does nothing if the queue is closed.
+    pub(crate) fn push(&self, task: task::Notified<T>) {
```

Comment:
Why is the `Result` return not needed anymore?

Reply:
I *think* I know, but I would like you to confirm it :)

Reply:
It's because the pushed task _may_ be a new task, and if the push fails for a new task, the caller needs to know so it can call `shutdown` on the task. With this PR, that's no longer necessary as binding the new task happens before pushing the notification, and if binding the task succeeded, then the runtime has been given responsibility for cleaning the new task up, even if submitting a notification for the task fails.

It is what I described with the following in the original PR text:

> The PR also introduces a conceptual change for the meaning of the `Notified` type — that type is now "just" a notification, and someone who holds a `Notified` is never responsible for cleaning up the task.